packages/web-react/src/types/modalComposed.ts

```diff
@@ -0,0 +1,42 @@
+import { ElementType, DetailedHTMLProps, HTMLAttributes, ComponentPropsWithRef } from 'react';
+import { ChildrenProps, StyleProps, ClickEvent } from './shared';
+
+export type ModalElementType = 'article' | 'form';
+
+export type DialogElementProps = DetailedHTMLProps<HTMLAttributes<HTMLDialogElement>, HTMLDialogElement>;
+
+export type ModalDialogHandlingProps = {
+  isOpen: boolean;
+  onClose: (event: ClickEvent) => void;
+};
+
+export type ModalDialogBaseProps<E extends ElementType = ModalElementType> = {
+  elementType?: E;
+} & ChildrenProps &
+  StyleProps;
+
+export type ModalDialogProps<E extends ElementType = ModalElementType> = ModalDialogBaseProps<E> &
+  Omit<ComponentPropsWithRef<E>, keyof ModalDialogBaseProps<E> | 'style' | 'className'>;
+
+export interface ModalBodyProps extends ChildrenProps, StyleProps {}
+
+export interface ModalHeaderProps extends ChildrenProps, StyleProps {
+  closeLabel?: string;
+}
+
+export interface ModalFooterProps extends ChildrenProps, StyleProps {
+  align?: 'left' | 'center' | 'right';
+  description?: string;
+}
+
+export interface ModalBaseProps
+  extends ChildrenProps,
+    Omit<DialogElementProps, 'style' | 'className'>,
+    StyleProps,
+    ModalDialogHandlingProps {
+  isExpandedOnMobile?: boolean;
+}
+
+export interface ModalComposedProps extends ModalBaseProps {}
```

Comment:
These two interfaces only extend each other and add no additional prop. So the `ModalBaseProps` should be the last one.

Reply:
fixed